model/project_vars.go

```diff
@@ -102,49 +103,47 @@ func FindMergedProjectVars(projectID string) (*ProjectVars, error) {
 	return projectVars, nil
 }
 
+// UpdateProjectVarsByValue searches all projects who have a variable set to the toReplace input parameter, and replaces all
+// matching project variables with the replacement input parameter. If dryRun is set to true, the update is not performed.
+// We return a list of keys that were replaced (or, the list of keys that would be replaced in the case that dryRun is true).
 func UpdateProjectVarsByValue(toReplace, replacement, username string, dryRun bool) (map[string][]string, error) {
 	catcher := grip.NewBasicCatcher()
-	matchingProjects, err := GetVarsByValue(toReplace)
+	matchingProjectVars, err := getVarsByValue(toReplace)
 	if err != nil {
 		catcher.Wrap(err, "fetching projects with matching value")
 	}
-	if matchingProjects == nil {
+	if matchingProjectVars == nil {
 		catcher.New("no projects with matching value found")
 	}
 	changes := map[string][]string{}
-	for _, project := range matchingProjects {
-		for key, val := range project.Vars {
+	for _, projectVars := range matchingProjectVars {
+		for key, val := range projectVars.Vars {
 			if val == toReplace {
 				if !dryRun {
-					originalVars := make(map[string]string)
-					for k, v := range project.Vars {
-						originalVars[k] = v
+					var beforeVars ProjectVars
+					err = util.DeepCopy(*projectVars, &beforeVars)
+					if err != nil {
+						catcher.Wrap(err, "copying project variables")
 					}
 					before := ProjectSettings{
-						Vars: ProjectVars{
-							Id:   project.Id,
-							Vars: originalVars,
-						},
+						Vars: beforeVars,
 					}
 
-					project.Vars[key] = replacement
-					_, err = project.Upsert()
+					projectVars.Vars[key] = replacement
+					_, err = projectVars.Upsert()
 					if err != nil {
-						catcher.Wrapf(err, "overwriting variables for project '%s'", project.Id)
+						catcher.Wrapf(err, "overwriting variables for project '%s'", projectVars.Id)
```

Comment:
Also should continue here -- we shouldn't log that the project was modified if it wasn't.